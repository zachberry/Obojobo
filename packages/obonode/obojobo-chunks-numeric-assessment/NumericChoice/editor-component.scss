@import '~styles/includes';

$lighten-color-highlight: lighten($color-highlight, 19);

.numeric-choice {
	$color-border: #dfdfdf;

	position: relative;
	padding-bottom: 1rem;
	border-bottom: 1px solid $color-border;

	&.is-selected {
		margin: -0.1em -0.1em 0 -0.1em;
		border: 1px solid $color-highlight;
		background-color: $lighten-color-highlight;
	}

	.correct-button {
		$correct-font: 'Segoe UI Symbol', arial, sans-serif;

		position: absolute;
		top: 2.35em;
		left: 1rem;
		font-size: inherit;
		background: transparent;
		font-family: $correct-font;
		border: 1px solid $color-border;
		border-radius: $dimension-rounded-radius * 0.75;
		width: 1.7em;
		height: 2.25em;
		cursor: pointer;

		&.is-correct {
			color: $color-correct;
		}

		&.is-not-correct {
			color: $color-incorrect;
		}
	}

	.numeric-feedback-editor {
		margin: 0 3.5rem;
	}
}
