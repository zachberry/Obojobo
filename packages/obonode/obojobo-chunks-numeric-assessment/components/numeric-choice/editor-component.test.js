import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'

import NumericChoice from './editor-component'

describe('NumericChoice Editor Node', () => {
	test('NumericChoice component', () => {
		const component = renderer.create(
			<NumericChoice
				node={{
					data: {
						get: () => {
							return {}
						}
					},
					nodes: []
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericChoice component correct choice', () => {
		const component = renderer.create(
			<NumericChoice
				node={{
					data: {
						get: () => {
							return { score: 100 }
						}
					},
					nodes: { size: 2 }
				}}
			/>
		)
		const tree = component.toJSON()

		expect(tree).toMatchSnapshot()
	})

	test('NumericChoice component deletes itself', () => {
		const editor = {
			removeNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericChoice
				node={{
					key: 'mockKey',
					nodes: [{ key: 'child1' }, { key: 'child2' }],
					data: {
						get: () => {
							return {}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('button')
			.at(1)
			.simulate('click')

		const tree = component.html()

		expect(editor.removeNodeByKey).toHaveBeenCalled()
		expect(tree).toMatchSnapshot()
	})

	test('NumericChoice component adds feedback', () => {
		const editor = {
			insertNodeByKey: jest.fn()
		}

		const component = mount(
			<NumericChoice
				node={{
					key: 'mockKey',
					nodes: [{ key: 'child1' }, { key: 'child2' }],
					data: {
						get: () => {
							return {}
						}
					}
				}}
				editor={editor}
			/>
		)

		component
			.find('button')
			.at(0)
			.simulate('click')

		expect(editor.insertNodeByKey).toHaveBeenCalled()
	})
})
