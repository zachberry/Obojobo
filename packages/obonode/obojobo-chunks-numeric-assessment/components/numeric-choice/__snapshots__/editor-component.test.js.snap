// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumericChoice Editor Node NumericChoice component 1`] = `
<div
  className="numeric-choice  is-not-selected"
>
  <div
    contentEditable={false}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center pad obojobo-draft--components--button alt-action is-not-dangerous align-center add-feedback-btn"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        Add Feedback
      </button>
    </div>
  </div>
  <div
    contentEditable={false}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
  </div>
</div>
`;

exports[`NumericChoice Editor Node NumericChoice component correct choice 1`] = `
<div
  className="numeric-choice  is-not-selected"
>
  <div
    contentEditable={false}
  >
    <div
      className="obojobo-draft--components--button is-not-dangerous align-center delete-button"
    >
      <button
        className="button"
        onClick={[Function]}
      >
        ×
      </button>
    </div>
  </div>
</div>
`;

exports[`NumericChoice Editor Node NumericChoice component deletes itself 1`] = `"<div class=\\"numeric-choice  is-not-selected\\"><div contenteditable=\\"false\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center pad obojobo-draft--components--button alt-action is-not-dangerous align-center add-feedback-btn\\"><button class=\\"button\\">Add Feedback</button></div></div><div contenteditable=\\"false\\"><div class=\\"obojobo-draft--components--button is-not-dangerous align-center delete-button\\"><button class=\\"button\\">×</button></div></div></div>"`;
